memory-viz/bin/cli.js

```diff
@@ -7,26 +7,31 @@ const { program } = require("commander");
 const { json } = require("node:stream/consumers");
 
 function parseFilePath(input) {
-    if (input) {
-        return pathExists(input, `File`);
-    } else {
-        return undefined;
+    const fullPath = path.resolve(process.cwd(), input);
+    if (fs.existsSync(fullPath, "File")) {
+        return fullPath;
     }
+    console.error(`Error: File ${fullPath} does not exist.`);
+    process.exit(1);
 }
 
 function parseOutputPath(input) {
-    return pathExists(input, `Output path`);
-}
-
-// helper function for parsing paths
-function pathExists(inputPath, errMsg) {
-    const fullPath = path.resolve(process.cwd(), inputPath);
-    if (!fs.existsSync(fullPath)) {
-        console.error(`Error: ${errMsg} ${fullPath} does not exist.`);
+    if (isPathDirectory(input)) {
+        console.error(`Error: Output path ${input} must be a file.`);
         process.exit(1);
-    } else {
-        return fullPath;
     }
+    const parsedPath = path.parse(input);
+    const outputDir = parsedPath.dir || ".";
+    const filename = `${parsedPath.name}.svg`;
+    if (!fs.existsSync(outputDir)) {
+        console.error(`Error: Output directory ${outputDir} does not exist.`);
+        process.exit(1);
+    }
+    return path.join(outputDir, filename);
+}
+
+function isPathDirectory(rawOutputPath) {
```

Comment:
This function isn't quite robust enough; consider, for example, someone who puts in a path `./hello` where `hello` is an existing directory.

You should figure out how to use Node's built-in functions to to tell whether an existing path refers to a directory or not.